ravem/indico_ravem/controllers.py

```diff
@@ -25,58 +30,90 @@ def _check_access(self):
+    def get_event_vc_room(self, _id):
+        event_vc_room = VCRoomEventAssociation.find_one(id=_id)
+        if not event_vc_room:
+            raise NotFound(_("Event VC Room not found for id {id}").format(id=_id))
+        if not event_vc_room.link_object:
+            raise IndicoError(
+                _("Event VC Room ({id}) is not linked to anything").format(id=_id)
+            )
+        return event_vc_room
 
-        if not self.event_vc_room.link_object:
-            raise IndicoError(_("Event VC Room ({id}) is not linked to anything").format(id=id_))
+    def _process_args(self):
+        _id = request.view_args['event_vc_room_id']
+        self.event_vc_room = self.get_event_vc_room(_id)
 
         event_id = int(request.view_args['confId'])
         event = self.event_vc_room.link_object.event
         if not event:
-            raise IndicoError(_("Event VC Room ({id}) does not have an event").format(id=id_))
-
+            raise IndicoError(
+                _("Event VC Room ({id}) does not have an event").format(id=_id)
+            )
         if event.id != event_id:
-            raise IndicoError(_("Event VC Room ({id}) does not have an event with the id {conf.id}")
-                              .format(id=id_, conf=event))
-
-        room = self.event_vc_room.link_object.room if self.event_vc_room.link_object else None
-        if not room:
-            raise IndicoError(_("Event VC Room ({id}) is not linked to an event with a room").format(id=id_))
-
-        self.room_name = room.generate_name()
-        self.room_special_name = room.name
-
-        if not self.room_name:
-            raise IndicoError(_("Event VC Room ({id}) is not linked to an event with a valid room").format(id=id_))
-
-        if not self.room_special_name:
-            self.room_special_name = self.room_name
+            raise IndicoError(
+                _(
+                    "Event VC Room ({id}) does not have an event with the id {conf.id}"
+                ).format(id=_id, conf=event)
+            )
+
+        self.room = (
+            self.event_vc_room.link_object.room
+            if self.event_vc_room.link_object
+            else None
+        )
+        if not self.room:
+            raise IndicoError(
+                _("Event VC Room ({id}) is not linked to an event with a room").format(
+                    id=_id
+                )
+            )
+        if not self.room.name:
+            raise IndicoError(
+                _(
+                    "Event VC Room ({id}) is not linked to an event with a valid room"
+                ).format(id=_id)
+            )
 
 
 class RHRavemRoomStatus(RHRavemBase):
     def _process(self):
+        if self.event_vc_room.vc_room.type == 'vidyo':
+            return {
+                'success': False,
+                'reason': 'unsupported',
+                'message': '<a href="http://cern.ch/go/6SNm">Vidyo is no longer supported</a> '
```

Comment:
I really hate returning a link here and feeding it directly to the client. But I'm not sure if it's worth complicating?
I also noticed this plugin is catching errors and converting them into a status 200 with a JSON flag `success`. Some of this boilerplate code could be simplified with actual exceptions.

Reply:
Seems fine to me... and with all the old jquery code it's usually easier to return anything expected as 200 and anything unexpected with an actual error code...

Reply:
Not sure about the redirector though... I'd at least use the https URL!

Reply:
do we need the trailing space in the message?

Reply:
No, both of your last two comments were my bad.